service/history/api/recordactivitytaskheartbeat/api.go

```diff
@@ -99,6 +99,11 @@ func Invoke(
 				return nil, consts.ErrActivityTaskNotFound
 			}
 
+			// update worker identity if available
```

Comment:
The change LGTM. 

Though does it mean for the given worker, its identity could change after picking an activity up? like it could change any time?
If that's the case, I guess don't know how to reason about this field...

Reply:
Even though the last worker identity is updated in the pending activity task info internally, we send this to client only if the `StartedIdentity` in activity info is also empty. So we sent the identity from the heartbeat only if the worker had not set the identity when picking up the task.